py/desispec/parallel.py

```diff
@@ -424,6 +432,29 @@ def _redirect(out_to, err_to):
             ch.setFormatter(formatter)
             logger.addHandler(ch)
 
+    def _combine_individual_outputs(to, nproc=None):
+        """
+        Combine individual {to}_{n} files into a single {to} file
+
+        If nproc is specified, assume there are exactly that many per-rank
+        files; otherwise glob to find out what is there.
+        """
+        if nproc is None:
+            individual_files = sorted(glob.glob(_rank_filename(to, '*')))
+        else:
+            individual_files = [_rank_filename(to, p) for p in range(nproc)]
+
+        with open(to, "w") as outfile:
+            for p, fname in enumerate(individual_files):
+                outfile.write("================ Start of Process {} ================\n".format(p))
+                with open(fname) as infile:
```

Comment:
This already existed in the original implementation, but when `nproc` is specified should we check for the existence of the files before trying to open them? They _should_ all be there, so maybe crashing on a missing rank's log file would help point to a processing issue. I don't have strong feelings either way, but I noticed it so I wanted to flag it for discussion.

Reply:
Good question. A missing per-rank file would indicate a core logic bug or a basic I/O failure so I think crashing here would be the right thing to do, not just merging whatever it can find and moving on.  Pre-checking for the file and printing an error message would be more code for approximately the same result as just letting Python raise a FileNotFound error with the missing filename, so I'm going to leave this as-is.